app/src/main/java/com/vultisig/wallet/ui/models/send/GasSettingsViewModel.kt

```diff
@@ -123,9 +123,14 @@ internal class GasSettingsViewModel @Inject constructor(
     }
 
     private fun loadEthData(chain: Chain, spec: BlockChainSpecificAndUtxo){
-        val specific = spec.blockChainSpecific as BlockChainSpecific.Ethereum
-
-        val gasLimit = specific.gasLimit
+        var gasLimit = if (gasLimitState.text.isEmpty()) {
+            val specific = spec.blockChainSpecific as BlockChainSpecific.Ethereum
+            specific.gasLimit.also {
+                gasLimitState.setTextAndPlaceCursorAtEnd(specific.gasLimit.toString())
```

Comment:
Because when a user saves the gas limit, closes it, and reopens it, it always displays 23000. However, in the backend, it correctly saves the value the user modified